scorecards.go

```diff
@@ -0,0 +1,124 @@
+package opslevel
+
+import (
+	"fmt"
+)
+
+type ScorecardId struct {
+	Aliases []string `graphql:"aliases"`
+	Id      ID       `graphql:"id"`
+}
+
+type Scorecard struct {
+	ScorecardId
+
+	Categories     []string `graphql:"categories"`
+	Checks         []string `graphql:"checks"`
+	Description    string   `graphql:"description"`
+	Filter         string   `graphql:"filter"`
+	Href           string   `graphql:"href"`
+	Levels         []string `graphql:"levels"`
+	Name           string   `graphql:"name"`
+	Owner          string   `graphql:"owner"`
+	PassingChecks  int      `graphql:"passingChecks"`
+	ServiceCount   int      `graphql:"serviceCount"`
+	ServicesReport string   `graphql:"servicesReport"`
+	Slug           string   `graphql:"slug"`
+	TotalChecks    int      `graphql:"totalChecks"`
+}
+
+type ScorecardConnection struct {
+	Nodes      []Scorecard `graphql:"nodes"`
+	PageInfo   PageInfo    `graphql:"pageInfo"`
+	TotalCount int         `graphql:"totalCount"`
+}
+
+type ScorecardInput struct {
+	Name        string          `graphql:"name" json:"name"`
+	Description string          `graphql:"description" json:"description,omitempty"`
```

Comment:
Just doing some early code review - on "Input" types when a field in the API is nullable we need to use pointers so the graphql framework will generate it correctly because you need 3 states - nil, empty string, filled string.  With just a string (value type) you only have 2 states - empty string, filled string.   

The `json` struct tag omitempty also works differently when its `string` vs `*string`
for `string` the omitempty will omit the field from the json when its the empty string state
for `*string` the omitempty will omit the field from the json when its the nil state but empty string state will still include the field in the json it will just be an empty string value which effectively works as an "unset the field" during an "update" mutation.

Its VERY complicated and it probably the hardest thing about making "Input" types.

Reply:
Thanks - I'll update this.